File: TOI-Zero_68/A2/032/032.py
n, m = map(int, input().split())

arr = [[0 for _ in range(m + 3)] for _ in range(n + 3)]

q = int(input())
for _ in range(q):
    a, b = map(int, input().split())
    arr[a + 1][b + 1] += 1

ma = 0

for i in range(1, n + 1):
    for j in range(1, m + 1):
        if arr[i][j] >= 1:
            continue
        ma = max(ma, arr[i][j] + arr[i - 1][j] + arr[i + 1][j] +
                 arr[i][j - 1] + arr[i][j + 1] +
                 arr[i - 1][j - 1] + arr[i + 1][j - 1] +
                 arr[i - 1][j + 1] + arr[i + 1][j + 1])

print(ma)

File: TOI-Zero_68/A2/032/032.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
	int n,m;
	cin>>n>>m;
	int arr[n+3][m+3];
	for(int i=0;i<n+3;i++){
		for(int j=0;j<m+3;j++){
			arr[i][j]=0;
		}
	}
	int q;
	cin>>q;
	while(q--){
		int a,b;
		cin>>a>>b;
		arr[a+1][b+1]++;
	}
	int ma=0;
	for(int i=1;i<=n;i++){
		for(int j=1;j<=m;j++){
			if(arr[i][j]>=1) continue;
			ma=max(ma,arr[i][j]+arr[i-1][j]+arr[i+1][j]+arr[i][j-1]+arr[i][j+1]+arr[i-1][j-1]+arr[i+1][j-1]+arr[i-1][j+1]+arr[i+1][j+1]);
		}
	}
	cout<<ma;
}
